### PR TITLE
cli: introduce edit-validation-sets

### PR DESCRIPTION
**Pull request by sergiusens**
- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `./runtests.sh static`?
- [ ] Have you successfully run `./runtests.sh tests/unit`?

-----
This is roughly based on edit-collaborators

### CODE REVIEW

**Review by cmatsuoka — APPROVED** [review]
Implementation looks sensible, +1.

File: snapcraft/cli/assertions.py [hunk 1]
```diff
@@ -204,6 +229,96 @@ def list_validation_sets(name, sequence):
...
+        )
+        store_client.post_validation_sets(signed_validation_sets=signed_validation_sets)
+
+
+def _edit_validation_sets(validation_sets: str) -> Dict[str, Any]:
+    """Spawn an editor to modify the validation-sets."""
+    editor_cmd = os.getenv("EDITOR", "vi")
+
+    with tempfile.NamedTemporaryFile() as ft:
+        ft.close()
```
Comment (cjp256) [inline snapcraft/cli/assertions.py:292]: Perhaps just write to it then close it.

File: snapcraft/cli/assertions.py [hunk 2]
```diff
@@ -204,6 +229,96 @@ def list_validation_sets(name, sequence):
...
+    cmdline = ["snap", "sign"]
+    if key_name:
+        cmdline += ["-k", key_name]
+    snap_sign = subprocess.Popen(
+        cmdline, stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE
+    )
+    signed_assertion, err = snap_sign.communicate(input=json.dumps(assertion).encode())
+    if snap_sign.returncode != 0:
+        echo.exit_error(
+            brief=details_from_command_error(
```
Comment (cjp256) [inline snapcraft/cli/assertions.py:312]:
```suggestion
            brief="Failed to sign assertion."
            details=details_from_command_error(...)
```

File: tests/unit/commands/test_edit_validation_sets.py [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+            '{"name": "snap-name-2", "id": "XXSnapIDForXSnapName2XXXXXXXXXXX"}], '
+            '"timestamp": "2020-10-29T16:36:56Z", "type": "validation-set", '
+            f'"series": "16"}}\n\nSIGNED{key_name}'
+        ).encode()
+    )
+    fake_snap_sign.assert_called_once_with(
+        validation_sets_payload.assertions[0].marshal(), key_name=key_name,
+    )
+    assert result.exit_code == 0
+    assert result.output.strip() == ""
```
Comment (cjp256) [inline tests/unit/commands/test_edit_validation_sets.py:207]: Perhaps add a test case for a signing failure case.

**Review by cjp256 — APPROVED** [review]

File: snapcraft/cli/assertions.py (hunk 1 above)
Comment (sergiusens) [reply snapcraft/cli/assertions.py:292]: I don't understand this comment, but we can iterate on this one if you want

**Pull request merged**